nexus/src/db/datastore.rs

```diff
@@ -1938,6 +1943,89 @@ impl DataStore {
                 ))
             })
     }
+
+    pub async fn users_builtin_list_by_name(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<UserBuiltin> {
+        use db::schema::user_builtin::dsl;
+        opctx.authorize(authz::Action::ListChildren, authz::FLEET)?;
+        paginated(dsl::user_builtin, dsl::name, pagparams)
+            .select(UserBuiltin::as_select())
+            .load_async::<UserBuiltin>(self.pool_authorized(opctx)?)
+            .await
+            .map_err(|e| {
+                public_error_from_diesel_pool(
+                    e,
+                    ResourceType::User,
+                    LookupType::Other("Listing All".to_string()),
+                )
+            })
+    }
+
+    pub async fn user_builtin_fetch(
+        &self,
+        opctx: &OpContext,
+        name: &Name,
+    ) -> LookupResult<UserBuiltin> {
+        use db::schema::user_builtin::dsl;
+        opctx.authorize(authz::Action::ListChildren, authz::FLEET)?;
+        dsl::user_builtin
+            .filter(dsl::name.eq(name.clone()))
+            .select(UserBuiltin::as_select())
+            .first_async::<UserBuiltin>(self.pool_authorized(opctx)?)
+            .await
+            .map_err(|e| {
+                public_error_from_diesel_pool(
+                    e,
+                    ResourceType::User,
+                    LookupType::ByName(name.as_str().to_owned()),
+                )
+            })
+    }
+
+    /// Load built-in users into the database
+    pub async fn load_builtin_users(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<(), Error> {
+        use db::schema::user_builtin::dsl;
+
+        opctx.authorize(authz::Action::Modify, authz::FLEET)?;
+
+        let builtin_users = [
+            // Note: "db_init" is also a builtin user, but that one by necessity
+            // is created with the database.
+            &*authn::USER_SAGA_RECOVERY,
+            &*authn::USER_TEST_PRIVILEGED,
+            &*authn::USER_TEST_UNPRIVILEGED,
```

Comment:
Do we want to conditionally include these?

Reply:
I think you're right that at some point, we'll want to leave them out of some configurations.  But right now, USER_TEST_PRIVILEGED is necessary to do anything useful with oxapi_demo or the console, even if you're not in `#[cfg(test)]`.  Once we add authz to the rest of the endpoints, you really won't be able to do anything useful with Nexus without these.  We will need to revisit this as part of initial rack setup, I think.

Reply:
On console we use a [DB seeding script](https://github.com/oxidecomputer/console/blob/main/tools/populate_omicron_data.sh) for local dev and for the QA instance deployed to GCP. Maybe these users could eventually move into something like that rather than a `#[cfg(test)]`.

Reply:
It might make sense to add a comment/create an appropriately tagged issue to track that?

Reply:
@teisenbe Sure thing.  I had a TODO-security for this (which I'm hoping we'll audit before we ship...but we don't track that either), but it got moved around and no longer really covers it.  Moved and clarified in af85c839.

@david-crespo Longer-term, another option would be to officially support local users and add "create" endpoints for them.  Maybe these would be better modeled as service accounts?  Then we'd just have the test suite create the ones it wants.  For developers playing around, they could create one with oxapi_demo.  But...with what creds would they create one?  Anyway I'd definitely like to move that discussion to a follow-on PR.